x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/api/ml_models/ml_trained_models_logic.ts

```diff
@@ -0,0 +1,169 @@
+  },
+  connect: {
+    actions: [
+      MLModelsApiLogic,
+      [
+        'apiError as getModelsApiError',
+        'apiReset as getModelsApiReset',
+        'apiSuccess as getModelsApiSuccess',
+        'makeRequest as makeGetModelsRequest',
+      ],
+      MLModelsStatsApiLogic,
+      [
+        'apiError as getModelsStatsApiError',
+        'apiReset as getModelsStatsApiReset',
+        'apiSuccess as getModelsStatsApiSuccess',
+        'makeRequest as makeGetModelsStatsRequest',
+      ],
+    ],
+    values: [
+      MLModelsApiLogic,
+      [
+        'apiStatus as modelsApiStatus',
+        'error as modelsApiError',
+        'status as modelsStatus',
+        'data as modelsData',
+      ],
+      MLModelsStatsApiLogic,
+      [
+        'apiStatus as modelStatsApiStatus',
+        'error as modelsStatsApiError',
+        'status as modelStatsStatus',
+        'data as modelStatsData',
+      ],
+    ],
+  },
+  listeners: ({ actions, values }) => ({
+    getModelsApiError: (error) => {
+      actions.apiError(error);
+    },
+    getModelsApiSuccess: () => {
+      if (!values.data) return;
+      actions.apiSuccess(values.data);
+    },
+    getModelsStatsApiError: (error) => {
+      if (values.modelsApiError) return;
+      actions.apiError(error);
+    },
+    getModelsStatsApiSuccess: () => {
+      if (!values.data) return;
+      actions.apiSuccess(values.data);
+    },
+    apiReset: () => {
+      actions.getModelsApiReset();
+      actions.getModelsStatsApiReset();
+    },
+    makeRequest: () => {
+      actions.makeGetModelsRequest(undefined);
+      actions.makeGetModelsStatsRequest(undefined);
+    },
+  }),
+  path: ['enterprise_search', 'api', 'ml_trained_models_api_logic'],
+  selectors: ({ selectors }) => ({
+    data: [
+      () => [selectors.modelsData, selectors.modelStatsData],
+      (
+        modelsData: TrainedModelsApiLogicValues['modelsData'],
+        modelStatsData: TrainedModelsApiLogicValues['modelStatsData']
+      ): TrainedModel[] | null => {
+        if (!modelsData) return null;
+        if (!modelStatsData) return modelsData;
+        const statsMap: Record<string, MlTrainedModelStats> =
+          modelStatsData.trained_model_stats.reduce((map, value) => {
+            if (value.model_id) {
+              map[value.model_id] = value;
+            }
+            return map;
+          }, {} as Record<string, MlTrainedModelStats>);
+        return modelsData.map((modelConfig) => {
+          const modelStats = statsMap[modelConfig.model_id];
+          return {
+            ...modelConfig,
+            ...(modelStats ?? {}),
+          };
+        });
+      },
+    ],
+    error: [
+      () => [selectors.modelsApiStatus, selectors.modelStatsApiStatus],
+      (
+        modelsApiStatus: TrainedModelsApiLogicValues['modelsApiStatus'],
+        modelStatsApiStatus: TrainedModelsApiLogicValues['modelStatsApiStatus']
+      ) => {
+        if (modelsApiStatus.error) return modelsApiStatus.error;
```

Comment:
question?: is `modelsApiStatus.error` is the error object?

Reply:
yes, I modeled this to follow the [api logic interface](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/public/applications/shared/api_logic/create_api_logic.ts#L74)